### ISSUE TITLE
update for newer plots

### ISSUE THREAD

**Issue by jverzani** [jverzani/SymPy.jl#73]
Write recipes...

**Comment by jverzani** [comment]
This is somewhat pressing. For now, the plotting feature has been deprecated so that SymPy will compile. It might be best to move the plotting into a package, as maintaining 0.3 compatability would be nice, but my suspicions are not possible using RecipesBase.

**Comment by jverzani** [comment]
Okay, here is what I have tried for Plots v"0.7.0":

```
using RecipesBase
@recipe f{T<:Sym}(::Type{T}, v::T) = lambdify(v)

@recipe function f(ex1::Sym, ex2::Sym,  a::Real, b::Real) # for 2, 3D parametric plots
    ts = linspace(a, b, 251)
    map(lambdify(ex1), ts), map(lambdify(ex2), ts)
end

@recipe function f(ex1::Sym, ex2::Sym,  ex3::Sym, a::Real, b::Real)
    ts = linspace(a, b, 251)
    map(lambdify(ex1), ts), map(lambdify(ex2), ts), map(lambdify(ex3), ts)
end

@recipe function f(ex::Vector{Sym}, a::Real, b::Real) # for contour plots
    (map(lambdify, ex), a, b)
end

```

I don't think the last three are the correct way to do this.

Also, I can't seem to get `quiver` to work as it did before. Is there a new invocation?

**Comment by jverzani** [comment]
Cool, it is working now for contours and parametric plots. Thanks. Does quiver work though in general? I'm never sure of the syntax, but the example code here https://github.com/tbreloff/Plots.jl/issues/262 throws an error for me.

**Comment by jverzani** [comment]
Wait, didn't test this one:

```
using SymPy, RecipesBase
@recipe f{T<:Sym}(::Type{T}, v::T) = lambdify(v)
@vars x
plot([x, -x], 0,1)
```

That complains:

```
WARNING: [a,b] concatenation is deprecated; use [a;b] instead
 in depwarn at deprecated.jl:73
 in oldstyle_vcat_warning at /Applications/Julia-0.4.5.app/Contents/Resources/julia/lib/julia/sys.dylib
 in vect at abstractarray.jl:32
while loading no file, in expression starting on line 0
ERROR: DivideError: integer division error
 [inlined code] from /Users/verzani/.julia/v0.4/Plots/src/series_new.jl:104
 in __apply_recipe#70__ at /Users/verzani/.julia/v0.4/RecipesBase/src/RecipesBase.jl:235
 in _plot! at /Users/verzani/.julia/v0.4/Plots/src/plot.jl:234
 in plot at /Users/verzani/.julia/v0.4/Plots/src/plot.jl:52
```

Whereas

```
plot([x->x, x->-x], 0, 1)
```

Works as expected.

**Comment by tbreloff** [comment]
This works for me:

```julia
julia> using Plots

julia> using SymPy, RecipesBase

julia> @recipe f(::Type{Sym}, v::Sym) = lambdify(v)
apply_recipe (generic function with 65 methods)

julia> @recipe f{S<:AbstractVector{Sym}}(::Type{S}, ss::S) = Function[lambdify(s) for s in ss]
apply_recipe (generic function with 66 methods)

julia> @vars x
(x,)

julia> plot([x, -x], 0,1)
[Plots.jl] Initializing backend: pyplot
```

I'd like to eventually handle vectors and matrices of type recipes automatically, but for now you have to define the conversion yourself.

As for the quiver, I just pushed a fix to dev... that example works for me.

**Comment by jverzani** [comment]
Closed with 5938ee4ea36c36e77250304a640649034cf91651
